### PR TITLE
Sign-In

### PR DESCRIPTION
**Pull request by AlaaTaima**
#10

### CODE REVIEW

**Comment by Fatmasiam** [comment]
add both the issue and the PR on the project board @AlaaTaima

File: src/containers/Firebase/firebase.js [hunk 1]
```diff
@@ -41,6 +41,12 @@ class Firebase {
   user = uid => this.db.doc(`users/${uid}`);
 
   users = () => this.db.collection('users');
+
+  // create collection for journals
+  // and each journal will be a document
+  journal = Jid => this.db.doc(`journals/${Jid}`);
```
Comment (yosefanajjar) [inline src/containers/Firebase/firebase.js:47]: Write descriptive variables names 
```suggestion
  journal = journalId => this.db.doc(`journals/${journalId}`);
```

File: src/containers/Firebase/firebase.js [hunk 2]
```diff
@@ -41,6 +41,12 @@ class Firebase {
   user = uid => this.db.doc(`users/${uid}`);
 
   users = () => this.db.collection('users');
+
+  // create collection for journals
+  // and each journal will be a document
```
Comment (yosefanajjar) [inline src/containers/Firebase/firebase.js:46]: This is an unnecessary comment

File: src/containers/SignIn/index.js [hunk 3]
```diff
@@ -25,19 +25,20 @@ class SignInForm extends React.Component {
...
-            localStorage.setItem('userId', res.user.uid);
-            this.setState({ loading: false });
-            history.push(ROUTES.HOME);
-          })
-          .catch(error => {
-            this.setState({ error, loading: false });
-          });
+        try {
+          const result = await firebase.doSignInWithEmailAndPassword(
+            values.email,
```
Comment (yosefanajjar) [inline src/containers/SignIn/index.js:36]: use destructuring here instead of using the dot notation

File: src/containers/SignIn/index.js [hunk 4]
```diff
@@ -25,19 +25,20 @@ class SignInForm extends React.Component {
...
+            values.password
+          );
+          await localStorage.setItem('userId', result.user.uid);
```
Comment (yosefanajjar) [inline src/containers/SignIn/index.js]: We don't really need to use await with local storage here, await is used **to wait for promises to be resolved** and local storage is not a promise.
- This line of code will only happen after the upper await finishes.

File: src/containers/SignIn/index.js [hunk 5]
```diff
@@ -25,19 +25,20 @@ class SignInForm extends React.Component {
...
+          this.setState({ loading: false });
+          await history.push(ROUTES.HOME);
```
Comment (yosefanajjar) [inline src/containers/SignIn/index.js]: - Same here we don't need to wait for the history push function
- Use destructuring here as will

**Review by Fatmasiam — APPROVED** [review]

**Pull request merged**